test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,328 @@
+                                                    double const prob_deletion,
+                                                    double const stddev = 0,
+                                                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t>> reads;
+    std::mt19937_64 gen{seed};
+    std::uniform_int_distribution<size_t> seeds{0};
+    std::normal_distribution<> dis_error_count{static_cast<double>(simulated_errors_), stddev};
+    std::uniform_real_distribution<double> probability{0.0, 1.0};
+
+    for (size_t i = 0; i < number_of_reads; ++i)
+    {
+        // TODO avoid mutating the same position multiple times
+        // simulate concrete error number or use normal distribution
+        uint8_t simulated_errors = (stddev == 0) ? simulated_errors_ :
+                                                   std::abs(std::round(dis_error_count(gen)));
+        std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(ref) - read_length - simulated_errors};
+        size_t rpos = random_pos(gen);
+        std::vector<alphabet_t> read_tmp{std::ranges::begin(ref) + rpos,
+                                         std::ranges::begin(ref) + rpos + read_length + simulated_errors};
+
+        for (uint8_t j = 0; j < simulated_errors; ++j)
+        {
+            double prob = probability(gen);
+            // Substitution
+            if (prob_insertion + prob_deletion < prob)
+            {
+                mutate_substitution(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Insertion
+            else if (prob_insertion < prob)
+            {
+                mutate_insertion(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Deletion
+            else
+            {
+                mutate_deletion(read_tmp, simulated_errors, seeds(gen));
+            }
+        }
+        read_tmp.erase(std::ranges::begin(read_tmp) + read_length, std::ranges::end(read_tmp));
+        reads.push_back(read_tmp);
+    }
+    return reads;
+}
+
+template <typename alphabet_t>
+std::vector<alphabet_t> generate_repeating_sequence(size_t const template_length = 5000,
+                                                    size_t const repeats = 20,
+                                                    double const template_fraction = 1,
+                                                    size_t const seed = 0)
+{
+    std::vector<alphabet_t> seq_template = generate_sequence<alphabet_t>(template_length, 0, seed);
+
+    // copy substrings of length len from seq_template mutate and concatenate them
+    size_t len = std::round(template_length * template_fraction);
+    uint8_t simulated_errors = 5;
+    len = (len + simulated_errors  > template_length) ? template_length - simulated_errors : len;
+
+    return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | view::persist | std::view::join;
```

Comment:
```suggestion
    return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | view::persist | std::view::join | std::ranges::to<std::vector>;
```
(the implicit conversion of view to ranges will be dropped in the near future so you need to ranges::to)